x/feemarket/post/fee.go

```diff
@@ -99,9 +101,22 @@ func (dfd FeeMarketDeductDecorator) PostHandle(ctx sdk.Context, tx sdk.Tx, simul
 		return ctx, err
 	}
 
-	err = state.Update(gas, params)
+	consensusParams, err := dfd.consensusParamsKeeper.Get(ctx)
 	if err != nil {
-		return ctx, errorsmod.Wrapf(err, "unable to update fee market state")
+		return ctx, sdkerrors.ErrKeyNotFound.Wrap(err.Error())
+	}
+
+	maxGas := params.MaxBlockUtilization
+	if consensusParams.Block.MaxGas < int64(params.MaxBlockUtilization) && consensusParams.Block.MaxGas >= 0 {
```

Comment:
This can happen if the fee-market params are updated separately from the ConsensusParams?

Reply:
Any reason why we wouldn't update `MaxBlockUtilization` (not saying we shld, j asking)

Reply:
Good point, that makes sense